test/functional/feature_icx_orderbook.py

```diff
@@ -0,0 +1,342 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014-2019 The Bitcoin Core developers
+# Copyright (c) DeFi Blockchain Developers
+# Distributed under the MIT software license, see the accompanying
+# file LICENSE or http://www.opensource.org/licenses/mit-license.php.
+"""Test ICX Orderbook."""
+
+from test_framework.test_framework import DefiTestFramework
+
+from test_framework.util import assert_equal
+
+from decimal import Decimal
+
+class ICXOrderbookTest (DefiTestFramework):
```

Comment:
Needs tests for icx_closeorder, icx_closeoffer and icx_getorder. Probably best to have a TX between the close tests, first time I tested close order it worked as the change used on make order which became of the Auth was then used as the input for the close order by chance and appeared to work.